src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -479,11 +516,10 @@ def schedule_unit(self, unit):
                  'lm_info'       : self._rm_lm_info,
                 }
 
-
         # allocation worked!  If the unit was tagged, store the node IDs for
         # this tag, so that later units can reuse that information
-        if tag:
-            self._tag_history[tag] = [node['uid'] for node in slots['nodes']]
+        if not dvm_hosts_list and tag is not None:
+            self._tag_history[tag] = [node['name'] for node in slots['nodes']]
```

Comment:
I would suggest to cover both cases here:
```py
if tag:
    if dvm_host_list:
        self._tag_history[tag] = dvm_hosts_list[tag]
    else: 
        self._tag_history[tag] = [node['name'] for node in slots['nodes']]
````

Reply:
here we deal with `tag` case that is not related to DVM, that's why cannot combine both of them here. When they all will be in `tags` this wouldn't be a case

Reply:
ack